Cargo.toml

```diff
@@ -39,13 +39,13 @@ prost = "^0.13.4"
 prost-derive = "^0.13.4"
 rand = "^0.8.5"
 regex = "^1.11.1"
-rustls = { version = "^0.23.27", optional = true }
+rustls = { version = "^0.23.27", default-features = false, optional = true }
 snap = { version = "^1.1.1", optional = true }
 serde = { version = "^1.0.219", features = ["derive"], optional = true }
 serde_json = { version = "^1.0.140", optional = true }
 tokio = { version = "^1.45.0", features = ["rt", "net", "time"], optional = true }
 tokio-util = { version = "^0.7.15", features = ["codec"], optional = true }
-tokio-rustls = { version = "0.26.2", optional = true }
+tokio-rustls = { version = "0.26.2", default-features = false, optional = true }
```

Comment:
Since it disables the default features, will it bring any breaking change for TLS use cases? Currently there is no integration test for TLS, so it's not quick to verify it.

Reply:
I took some time to reimplement this in a way that should not break compatibility. Does it look better?